File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/services/proxies/FieldProxy.java
package com.github.ol_loginov.heaplibweb.services.proxies;

import com.github.ol_loginov.heaplibweb.repository.heap.FieldEntity;
import com.github.ol_loginov.heaplibweb.repository.heap.HeapScope;
import lombok.RequiredArgsConstructor;
import org.netbeans.lib.profiler.heap.Field;
import org.netbeans.lib.profiler.heap.JavaClass;
import org.netbeans.lib.profiler.heap.Type;

@RequiredArgsConstructor
public class FieldProxy implements Field {
	private final FieldEntity entity;
	private final HeapScope scope;

	@Override
	public JavaClass getDeclaringClass() {
		return null;
	}

	@Override
	public String getName() {
		return entity.getName();
	}

	@Override
	public boolean isStatic() {
		return entity.getStaticFlag();
	}

	@Override
	public Type getType() {
		return scope.getTypes().findById(entity.getTypeId())
			.map(e -> (Type) new TypeProxy(e))
			.orElse(null);
	}
}

File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/services/proxies/FieldProxy.kt
package com.github.ol_loginov.heaplibweb.services.proxies

import com.github.ol_loginov.heaplibweb.repository.heap.FieldEntity
import com.github.ol_loginov.heaplibweb.repository.heap.HeapRepositories
import org.netbeans.lib.profiler.heap.Field
import org.netbeans.lib.profiler.heap.JavaClass
import org.netbeans.lib.profiler.heap.Type

class FieldProxy(
    private val entity: FieldEntity,
    private val heapRepositories: HeapRepositories
) : Field {
    override fun getDeclaringClass(): JavaClass = JavaClassProxy.wrap(heapRepositories.classes.findByIdSure(entity.declaringClassId), heapRepositories)
    override fun getName() = entity.nameId.toString()
    override fun isStatic() = entity.staticFlag
    override fun getType(): Type? = null
}